### ISSUE TITLE
what is difference between reg:logistic and binary:logistic?

### ISSUE THREAD

**Issue by billbargens** [dmlc/xgboost#521]
What's the difference between reg:logistic and binary:logistic? They all output probability and use logistic loss.

**Comment by tqchen** [comment]
There is no difference, except that the default evaluation metric for reg:logistic is rmse, as far as I recall

**Comment by billbargens** [comment]
What about the training data? 
I Once thought  I can only feed the training data with target label y to be {0 or 1} when i set the objective to binary:logistic while the y should be a   continuous value between 0 and 1 when i set the objective to reg:logistic?

**Comment by tqchen** [comment]
There is no difference in terms of training data. You can also feed in probabilistic values in ```[0,1]```

**Comment by billbargens** [comment]
I got it. Thank you!

**Comment by jukiewiczm** [comment]
Hello,
I have a question. How is it possible that you pass a continuous (probabilistic) value to the logistic objective (I mean, the label data)? I have never encountered this before. Are you able to point me how does the logistic objective look like in xgboost? I know that you can simply put continuous value to the standard logistic objective, but does it make any sense and have some scientific justification? I'm not trying to discredit the approach in any way, just trying to understand it.

**Comment by hengqujushi** [comment]
@jukiewiczm [https://pan.baidu.com/s/1c1SazZE](url)
